Bottom: af891520282ccac6ad997d3f372f2f912a9198d8
Top:    d07174e0aabd3c1ed87b55b5c6bb1ddfacb0759c
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-23 14:02:10 -0400

Refresh of of-soc-simple-rework

---

diff --git a/sound/soc/fsl/soc-of-simple.c b/sound/soc/fsl/soc-of-simple.c
index 2b32f64..5f859a8 100644
--- a/sound/soc/fsl/soc-of-simple.c
+++ b/sound/soc/fsl/soc-of-simple.c
@@ -132,7 +132,7 @@ int of_snd_soc_register_codec(struct snd_soc_codec_device *codec_dev,
 	of_soc->card.num_links = min(count, of_soc->card.num_links);
 	for (i = 0; i < of_soc->card.num_links; i++) {
 		of_soc->dai_link[i].name = dai[i].name;
-		of_soc->dai_link[i].codec_dai = dai;
+		of_soc->dai_link[i].codec_dai = &dai[i];
 	}
 	/* Now try to register the SoC device */
 	of_snd_soc_register_device(of_soc);
